dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/TasksRunner.java

```diff
@@ -168,6 +181,7 @@ private <T> T runTask(Task<T> task) throws MetadataDumperUsageException {
       else if (!task.handleException(e))
         logger.warn("Task failed: {}: {}", task, e.getMessage(), e);
       state.setTaskException(task, TaskState.FAILED, e);
+      MetricToErrorMap.put(task.getName(), e.getMessage());
```

Comment:
We do not have any unique guaranties for `tast.getName()`. You need to redesign this approach.